packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/refresh.md

```diff
@@ -1,39 +1,60 @@
+You refreshed a token when checking for an existing session in [Check for a session at startup](#check-for-a-session-at-startup). In this case you may want to catch any errors that occur.
+
 Update `refreshToken` in `ContentView` with the following code:
 
 ```swift
 func refreshToken() {
-  guard let authStateManager = authStateManager else {
-    showError(title: "Unable to Renew Token",
-              message: "Could not communincate with the Okta server.")
-    return
-  }
-
-  var tokenInfo = OKTIDToken.init(idTokenString: authStateManager.accessToken!)
-  let currentExpiry = tokenInfo!.expiresAt
-
-  authStateManager.renew() { authStateManager, error in
-    if let error = error {
-      showError(title: "Unable to Renew Token",
-      error: error)
-      return
-    }
-
-    if authStateManager != nil {
-      self.authStateManager = authStateManager
-    }
-
-    tokenInfo = OKTIDToken.init(idTokenString: (authStateManager?.accessToken!)!)
-
-    if tokenInfo!.expiresAt == currentExpiry {
-      showError(title: "Token Not Renewed",
-                message: "An uknown issue prevented renewing the token.")
-    } else {
-      authStateManager?.writeToSecureStorage()
-      statusText = "Token renewed."
-      showTokenInfo()
-    }
-  }
+   if let credential = Credential.default {
+      Task {
+         busy = true
+         do {
+            try await credential.refreshIfNeeded()
+         }
+         catch {
+            showError(title: "Unable to Refresh Token", error: error)
+         }
+         busy = false
+      }
+   } else {
+      showError(title: "Unable to Refresh Token", message: "An unknown issue prevented refreshing the token. Please try again.")
+   }
+}
+```
+
+Most of the code is for handling errors or controling the state of the busy view.
```

Comment:
```suggestion
Most of the code is for handling errors or controlling the state of the busy view.
```

Reply:
I hat it whon I meighk spalling errers. In the next commit.